### DISCUSSION TITLE
Change text/visibility of a command/button

### DISCUSSION THREAD

**Discussion by ghost** [VsixCommunity/Community.VisualStudio.Toolkit#505] [Q&A] [answered]
Hi, I need to change visibility / change the text of a button/menu command in runtime.

I've found this piece of documentation: https://learn.microsoft.com/en-us/visualstudio/extensibility/changing-the-appearance-of-a-command?view=vs-2022 

However, I wonder if community toolkit makes this process less painful than the official examples.

Thanks in advance.

**Comment by ErikEJ** [comment]
@pbanosdev There is some sample code in the docs here: https://learn.microsoft.com/en-us/visualstudio/extensibility/changing-the-text-of-a-menu-command?view=vs-2022

**Reply by ghost** [reply]
Thanks for the quick answer Erik. I'm aware of that documentation, but, for example, the structure of my command classes (Community Toolkit) don't have a constructor (step 4). So this instructions are not "compatible" with how extensions are built following Community Toolkit tutorials; 

https://learn.microsoft.com/en-us/visualstudio/extensibility/vsix/visual-studio-community-toolkit?view=vs-2022

So the main question here is, if I have a Community Toolkit project, how can I achieve the mentioned functionality?

**Comment by jamesc-skyward** [comment, marked as answer]
In your Command class that inherits from BaseCommand, you should have access to the Command property: https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/blob/2fac3c44fdeff50c93908fe943b34ab194e340e3/src/toolkit/Community.VisualStudio.Toolkit.Shared/Commands/BaseCommand.cs#L100

With this you should be able to change the text and visibility.

**Reply by ghost** [reply]
Hi! Thanks for the answer. 

It turned out to be easier than I thought, I was definitely over complicating things. Junior mistake, I guess. I've been assigned at work with developing an extension and it's the first time for me.  

Have a nice day.
